marlowe-cli/src/Language/Marlowe/CLI/Command/Parse.hs

```diff
@@ -208,18 +209,27 @@ readAssetIdEither s =
 
 
 
--- | Parser for `AddressAny`.
-parseAddressAny :: O.ReadM AddressAny
-parseAddressAny = O.eitherReader readAddressAnyEither
+-- | Parser for `AddressInEra era`.
+parseAddress :: IsShelleyBasedEra era => O.ReadM (AddressInEra era)
+parseAddress = O.eitherReader readAddressEither
 
 
--- | Parser for `AddressAny`.
-readAddressAnyEither :: String                    -- ^ The string to be read.
-                     -> Either String AddressAny  -- ^ Either the address or an error message.
-readAddressAnyEither s =
-  case deserialiseAddress AsAddressAny $ T.pack s of
+-- | Parser for `AddressInEra era`.
+readAddressEither :: forall era
+                   . IsShelleyBasedEra era
+                  => String                    -- ^ The string to be read.
+                  -> Either String (AddressInEra era)  -- ^ Either the address or an error message.
+readAddressEither s = do
+  era <- eraAsType
+  case deserialiseAddress (AsAddressInEra era) $ T.pack s of
     Nothing      -> Left "Invalid address."
     Just address -> Right address
+  where
+    eraAsType :: Either String (AsType era)
```

Comment:
I have to for sure pair with you @jhbertra so we can together read `HasTypeProxy` ;-)